assembly-main/src/assembly/bin/che.sh

```diff
@@ -482,42 +493,57 @@ function call_catalina {
   fi
 }
 
-function stop_che_server {
+stop_che_server () {
 
   if ! ${USE_DOCKER}; then
     echo -e "Stopping Che server running on localhost:${CHE_PORT}"
     call_catalina >/dev/null 2>&1
   else
-    echo -e "Stopping Che server running in docker container named ${GREEN}che${NC}."
+    echo -e "Stopping Che server running in docker container named ${GREEN}${CONTAINER}${NC}..."
 
-    DOCKER_EXEC="sudo service docker stop && /home/user/che/bin/che.sh stop"
-    "${DOCKER}" exec che ${DOCKER_EXEC} &>/dev/null 2>&1 || DOCKER_EXIT=$? || true
+    echo -e "Stopping docker container named ${GREEN}${CONTAINER}${NC}...."
+    DOCKER_EXEC="sudo service docker stop && sudo //home/user/che/tomcat/bin/shutdown.sh"
+    "${DOCKER}" exec ${CONTAINER} ${DOCKER_EXEC} || DOCKER_EXIT=$? || true
 
-    echo -e "Stopping docker container named ${GREEN}che${NC}."
-    "${DOCKER}" stop che &>/dev/null 2>&1 || DOCKER_EXIT=$? || true
+    "${DOCKER}" stop --time=10 ${CONTAINER} &>/dev/null 2>&1 || DOCKER_EXIT=$? || true
   fi
+  return
```

Comment:
While it looks that way it is still necessary to avoid interrupted scripts errors is the docker command is interrupted with a sigkill.  Without this the script terminates with syntax errors